plugin/src/main/java/org/opensearch/ml/action/model_group/SearchModelGroupTransportAction.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.action.model_group;
+
+import static org.opensearch.ml.action.handler.MLSearchHandler.wrapRestActionListener;
+import static org.opensearch.ml.settings.MLCommonsSettings.ML_COMMONS_VALIDATE_BACKEND_ROLES;
+import static org.opensearch.ml.utils.SecurityUtils.addUserBackendRolesFilter;
+import static org.opensearch.ml.utils.SecurityUtils.isAdmin;
+
+import lombok.extern.log4j.Log4j2;
+
+import org.opensearch.action.ActionListener;
+import org.opensearch.action.search.SearchRequest;
+import org.opensearch.action.search.SearchResponse;
+import org.opensearch.action.support.ActionFilters;
+import org.opensearch.action.support.HandledTransportAction;
+import org.opensearch.client.Client;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.inject.Inject;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.commons.authuser.User;
+import org.opensearch.ml.action.handler.MLSearchHandler;
+import org.opensearch.ml.common.transport.model_group.MLModelGroupSearchAction;
+import org.opensearch.ml.utils.RestActionUtils;
+import org.opensearch.tasks.Task;
+import org.opensearch.transport.TransportService;
+
+@Log4j2
+public class SearchModelGroupTransportAction extends HandledTransportAction<SearchRequest, SearchResponse> {
+    private MLSearchHandler mlSearchHandler;
+
+    Client client;
+    ClusterService clusterService;
+
+    private volatile boolean filterByEnabled;
+
+    @Inject
+    public SearchModelGroupTransportAction(
+        TransportService transportService,
+        ActionFilters actionFilters,
+        MLSearchHandler mlSearchHandler,
+        Client client,
+        Settings settings,
+        ClusterService clusterService
+    ) {
+        super(MLModelGroupSearchAction.NAME, transportService, actionFilters, SearchRequest::new);
+        this.mlSearchHandler = mlSearchHandler;
+        this.client = client;
+        this.clusterService = clusterService;
+        filterByEnabled = ML_COMMONS_VALIDATE_BACKEND_ROLES.get(settings);
```

Comment:
We can move this `filterByEnabled` consumer to a helper class to avoid duplicate code in every class.

Reply:
I tried it but it was not working as expectecd as this setting was to be injected into these constructors.

Reply:
This setting can be injected into these constructors but not have to, it can be encapsulated in a helper class and using a public method to return its value, the helper class can be created in MachineLearningPlugin.java and been injected to other transport action constructors. This is not a p0 issue but if got time, please have a try again and if it's not working, please reach out to me, we can dig this issue together.

Reply:
Okay got it. Will try to implement it. If anything, will reach out to you